sencha-workspace/SlateTasksStudent/app/controller/Dashboard.js

```diff
@@ -136,6 +148,49 @@ Ext.define('SlateTasksStudent.controller.Dashboard', {
         });
     },
 
+    onStudentSelectorSelect: function(combo, rec) {
+        var me = this,
+            studentCombo = me.getStudentSelectorCombo(),
+            params = me.getSearchParams(),
+            username;
+
+        if (!studentCombo) {
+            return;
+        }
+
+        // skip rerouting if student selection hasn't changed
+        username = studentCombo.getValue();
+        if (username === params.student) {
+            return;
+        }
+
+        // reroute page to reflect selected student
+        params.student = username;
+        location.search = Ext.Object.toQueryString(params);
+    },
+
+    onStudentSelectorBoxReady: function(combo) {
+        var params = this.getSearchParams();
+        combo.getStore().load();        
+        combo.setValue(params.student);
+
+        // reset section state
+        this.redirectTo('section/all');
```

Comment:
shouldn't this be done when a student is selected instead? Redirecting during initial rendering is a bad idea generally

Reply:
@themightychris This is done to ensure if you switch from one student to another the previous section selection is removed. Ie if you have student 1 English 1 to student 2 you need to make sure it goes back to section/all in case student 2 is not a member of English 1.

I tried implementing this in onStudentSelectorSelect but if I try to reset the route and the change the query string I was getting an error.

Reply:
@themightychris I moved the section resetter to after the user choose a new student and the following error appears: 'TypeError: cannot read property 'mousedown' of null' 

You were correct in saying if you turn off the Pause on Exception flag (or don't have the inspector working at all) it'll work just fine however I still haven't found a way to remove this error from popping up in the first place.